### PR TITLE
Fix #221: Introduce documentation of API and SPI

### LINKED ISSUE #221: Introduce documentation of API and SPI
Introduce a way to document API and SPI for better communication to downstream projects and easier management of backward comaptibility.

### PR DESCRIPTION
**Pull request by banterCZ**

### CODE REVIEW

File: enrollment-server-common/src/main/java/com/wultra/app/enrollmentserver/common/annotation/PublicApi.java [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package com.wultra.app.enrollmentserver.common.annotation;
+
+import java.lang.annotation.Documented;
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Target;
+
+/**
+ * Marker for interfaces intended to be called by downstream projects. Implementation may not be exposed by the core functionality.
```
Comment (banterCZ) [inline enrollment-server-common/src/main/java/com/wultra/app/enrollmentserver/common/annotation/PublicApi.java:25]: Description inspired by http://wiki.apidesign.org/wiki/APIvsSPI

File: enrollment-server-onboarding/src/main/java/com/wultra/app/onboardingserver/provider/OnboardingProvider.java [hunk 2]
```diff
@@ -26,6 +27,7 @@
  *
  * @author Roman Strobl, roman.strobl@wultra.com
  */
+@PublicSpi
```
Comment (banterCZ) [inline enrollment-server-onboarding/src/main/java/com/wultra/app/onboardingserver/provider/OnboardingProvider.java:30]: The very first usage.

File: enrollment-server-common/src/main/java/com/wultra/app/enrollmentserver/common/annotation/PublicApi.java [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+ * <p>
+ * New methods can be added.
+ * Those clients that used to call the previously existing elements, don't need to care about the new ones.
```
Comment (romanstrobl) [inline enrollment-server-common/src/main/java/com/wultra/app/enrollmentserver/common/annotation/PublicApi.java:28]: I am not sure I fully understand this sentence. Is this related to new methods? What are the elements exactly?

**Review by romanstrobl — APPROVED** [review]
In general, I really like the idea. Let's polish the JavaDoc a bit.
